operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -706,9 +769,24 @@ private void internalMakeRightDomainPresence() {
     }
 
     private StepAndPacket createDomainPlanSteps(Packet packet) {
+      if (containsAbortedEventData()) {
+        return new StepAndPacket(Step.chain(createEventStep(eventData), new TailStep()), packet);
+      }
+
       return new StepAndPacket(
-          createPopulatePacketServerMapsStep(createSteps()),
-          packet);
+          getEventStep(Step.chain(createPopulatePacketServerMapsStep(),  createSteps())), packet);
+    }
+
+    private Step getEventStep(Step next) {
+      if (eventData != null) {
+        Step step = Step.chain(createEventStep(eventData), next);
```

Comment:
There is no need to create `step` as a local variable, here. You could more cleanly simply return the created object. IntelliJ should have made this recommendation (look for the yellow bars on the right margin).

You might also consider using Optional. In this case, the entire method would be something like:

```
private Step getEventStep(final Step next) {
  return Optional.ofNullable(eventData).map(ed -> Step.chain(createEventStep(ed), next)).orElse(next);
}
```